File: loom-compiler-old/src/main/java/loom/compiler/ast/statements/DoWhileStmt.java
package loom.compiler.ast.statements;

import loom.compiler.ast.Expr;
import loom.compiler.ast.NodeVisitor;
import loom.compiler.ast.Stmt;
import loom.compiler.token.TokenSpan;

import java.util.List;

public class DoWhileStmt implements Stmt {
	public final List<Stmt> body;
	public final Expr condition;
	public final TokenSpan span;

	public DoWhileStmt(List<Stmt> body, Expr condition, TokenSpan span) {
		this.body = body;
		this.condition = condition;
		this.span = span;
	}

	@Override
	public <T> T accept(NodeVisitor<T> visitor) {
		return visitor.visitDoWhileStmt(this);
	}

	@Override
	public TokenSpan getSpan() {
		return span;
	}
} 

File: Frontend/Parser/AST/Statements/DoWhileStmt.cs
// Copyright (c) 2026.The Cloth contributors.
// 
// DoWhileStmt.cs is part of the Cloth Frontend.
// 
// Use, modification, and distribution of this file are governed by the
// license terms provided with the Cloth Compiler source distribution.

using FrontEnd.Parser.AST.Expressions;
using FrontEnd.Token;

namespace FrontEnd.Parser.AST.Statements;

public readonly record struct DoWhileStmt(Block Body, Expression Condition, TokenSpan Span);